x/emissions/keeper/whitelist.go

```diff
@@ -0,0 +1,308 @@
+// or (the parameter is true and topicWorkerWhitelist keyset has the (topicId, actor) key)
+func (k *Keeper) IsEnabledTopicWorker(ctx context.Context, topicId TopicId, actor ActorId) (bool, error) {
+	topicWhitelistEnabled, err := k.IsTopicWorkerWhitelistEnabled(ctx, topicId)
+	if err != nil {
+		return false, err
+	}
+	if topicWhitelistEnabled {
+		// If whitelist enabled check to see if actor is whitelisted
+		return k.IsWhitelistedTopicWorker(ctx, topicId, actor)
+	}
+	return true, nil
+}
+
+// An actor is topic reputer whitelisted if the topicWhitelistEnabled parameter is false
+// or (the parameter is true and topicReputerWhitelist keyset has the (topicId, actor) key)
+func (k *Keeper) IsEnabledTopicReputer(ctx context.Context, topicId TopicId, actor ActorId) (bool, error) {
+	topicWhitelistEnabled, err := k.IsTopicReputerWhitelistEnabled(ctx, topicId)
+	if err != nil {
+		return false, err
+	}
+	if topicWhitelistEnabled {
+		// If whitelist enabled check to see if actor is whitelisted
+		return k.IsWhitelistedTopicReputer(ctx, topicId, actor)
+	}
+	return true, nil
+}
+
+/// PERMISSIONS - Functions that determine if an actor has the ability to perform an action
+
+// Whitelist admins can update global whitelists including adding/removing from the global actor and whitelist admin lists
+func (k *Keeper) CanUpdateGlobalWhitelists(ctx context.Context, actor ActorId) (bool, error) {
+	return k.IsWhitelistAdmin(ctx, actor)
+}
+
+// Whitelist admins can update global parameters
+func (k *Keeper) CanUpdateParams(ctx context.Context, actor ActorId) (bool, error) {
+	return k.IsWhitelistAdmin(ctx, actor)
+}
+
+// Whitelist admins and topic creators can update topic whitelists
+// Updating the whitelist includes adding/removing from the whitelist and enabling/disabling the whitelist
+func (k *Keeper) CanUpdateTopicWhitelist(ctx context.Context, topicId TopicId, actor ActorId) (bool, error) {
+	topic, err := k.GetTopic(ctx, topicId)
+	if err != nil {
+		return false, err
+	}
+	if topic.Creator == actor {
+		return true, nil
+	}
+	return k.IsWhitelistAdmin(ctx, actor)
+}
+
+// An actor can create a topic if they are topic creator whitelisted
+// or if they are globally whitelisted
+func (k *Keeper) CanCreateTopic(ctx context.Context, actor ActorId) (bool, error) {
+	isTopicCreator, err := k.IsEnabledWhitelistedTopicCreator(ctx, actor)
+	if err != nil {
+		return false, err
+	}
+
+	if isTopicCreator {
+		return true, nil
+	}
+
+	return k.IsEnabledGlobalActor(ctx, actor)
+}
+
+// An actor can submit a worker payload if they are topic worker whitelisted
+// or if they are globally whitelisted
+func (k *Keeper) CanSubmitWorkerPayload(ctx context.Context, topicId TopicId, actor ActorId) (bool, error) {
+	has, err := k.IsEnabledTopicWorker(ctx, topicId, actor)
+	if err != nil {
+		return false, err
+	}
+	if has {
+		return true, nil
+	}
+	return k.IsEnabledGlobalActor(ctx, actor)
+}
```

Comment:
`isTopicCreator` -> `has`

Reply:
will change the latter to match the former, not the former to match the latter